sdk/src/main/scala/com/horizen/forge/Forger.scala

```diff
@@ -30,19 +30,33 @@ class Forger(settings: SidechainSettings,
   val timeoutDuration: FiniteDuration = settings.scorexSettings.restApi.timeout
   implicit val timeout: Timeout = Timeout(timeoutDuration)
 
-  @volatile private var forgingIsActive: Boolean = false
 
-  val consensusMillisecondsInSlot: Int = params.consensusSecondsInSlot * 1000
-  val forgingInitiator: Runnable = () => {
-    while (true) {
-      if (forgingIsActive) {
-        tryToCreateBlockNow()
+  private val consensusMillisecondsInSlot: Int = params.consensusSecondsInSlot * 1000
+  private def forgingInitiatorTimerTask: TimerTask = new TimerTask {override def run(): Unit = tryToCreateBlockNow()}
+  private var timerOpt: Option[Timer] = None
+
+  private def startTimer(): Unit = {
+    this.timerOpt match {
+      case Some(_) => log.info("Automatically forging already had been started")
+      case None => {
+        val newTimer = new Timer()
+        newTimer.scheduleAtFixedRate(forgingInitiatorTimerTask, 0, consensusMillisecondsInSlot)
```

Comment:
to avoid to always trigger the timer very close to slot boundaries, the initial execution moment must be chosen properly. For this reason let's calculate the initial delay needed to start the first execution in the middle of the first available "mid" slot and pass this initial delay to the scheduleAtFixedRate function